src/test/java/sg/dex/starfish/integration/keeper/DirectPurchaseTest.java

```diff
@@ -0,0 +1,65 @@
+package sg.dex.starfish.integration.keeper;
+
+import org.junit.Test;
+
+import com.oceanprotocol.squid.exceptions.EthereumException;
+import org.web3j.crypto.CipherException;
+import java.io.IOException;
+
+import java.math.BigInteger;
+import com.oceanprotocol.squid.models.Account;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import sg.dex.starfish.impl.squid.SquidAccount;
+import sg.dex.starfish.impl.squid.SquidService;
+import sg.dex.starfish.keeper.DirectPurchaseAdapter;
+import com.oceanprotocol.squid.api.OceanAPI;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DirectPurchaseTest {
+    private DirectPurchaseAdapter directPurchaseAdapter;
+    private String reference;
+    private BigInteger tokenAmount;
+    String receiverAddress;
+    String senderAddress;
+
+    public DirectPurchaseTest() throws IOException, CipherException {
+        directPurchaseAdapter = new DirectPurchaseAdapter();
+        reference = "0x000000000000a46daef00000000000";
+        tokenAmount = BigInteger.valueOf(10);
+        receiverAddress = "0x068Ed00cF0441e4829D9784fCBe7b9e26D4BD8d0";
+        senderAddress = "0x00bd138abd70e2f00903268f3db08f2d25677c9e";
+    }
+
+    @Test
+    public void Purchase() throws EthereumException {
```

Comment:
Naming: should be something like `testPurchase`. Avoid capitals for method names.

Reply:
Done